server/neptune/workflows/internal/deploy/terraform/state.go

```diff
@@ -39,6 +48,35 @@ func (n *StateReceiver) Receive(ctx workflow.Context, c workflow.ReceiveChannel,
 		}
 	}
 
+	if workflowState.Apply != nil &&
+		workflowState.Apply.Status == state.WaitingJobStatus &&
+		!reflect.ValueOf(workflowState.Apply.OnWaitingActions).IsZero() {
+		// update queue with information about current deployment pending confirm/reject action
+		infos := n.Queue.GetOrderedMergedItems()
+
+		revisionsSummary := n.Queue.GetQueuedRevisionsSummary()
+		state := github.CheckRunQueued
+		runLink := github.BuildRunURLMarkdown(deploymentInfo.Repo.GetFullName(), deploymentInfo.Commit.Revision, deploymentInfo.CheckRunID)
+		summary := fmt.Sprintf("This deploy is queued pending action on run for revision %s.\n%s", runLink, revisionsSummary)
+
+		for _, i := range infos {
+			request := notifier.GithubCheckRunRequest{
+				Title:   notifier.BuildDeployCheckRunTitle(i.Root.Name),
+				Sha:     i.Commit.Revision,
+				State:   state,
+				Repo:    i.Repo,
+				Summary: summary,
+			}
+
+			workflow.GetLogger(ctx).Debug(fmt.Sprintf("Updating action pending summary for deployment id: %s", i.ID.String()))
+			_, err := n.CheckRunCache.CreateOrUpdate(ctx, i.ID.String(), request)
```

Comment:
I think this might pose some non-determinism errors, see https://community.temporal.io/t/workflow-determinism/4027, but I could be wrong.  If it does, let's add some workflow versioning here